velox/functions/lib/aggregates/CentralMomentsAggregatesBase.h

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Aggregate.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::functions::aggregate {
+
+// Indices into RowType representing intermediate results of skewness and
+// kurtosis. Columns appear in alphabetical order.
+struct CentralMomentsIndices {
+  int32_t count;
+  int32_t m1;
+  int32_t m2;
+  int32_t m3;
+  int32_t m4;
+};
+constexpr CentralMomentsIndices kCentralMomentsIndices{0, 1, 2, 3, 4};
+
+struct CentralMomentsAccumulator {
+  int64_t count() const {
+    return count_;
+  }
+
+  double m1() const {
+    return m1_;
+  }
+
+  double m2() const {
+    return m2_;
+  }
+
+  double m3() const {
+    return m3_;
+  }
+
+  double m4() const {
+    return m4_;
+  }
+
+  void update(double value) {
+    double oldCount = count();
+    count_ += 1;
+    double oldM1 = m1();
+    double oldM2 = m2();
+    double oldM3 = m3();
+    double delta = value - oldM1;
+    double deltaN = delta / count();
+    double deltaN2 = deltaN * deltaN;
+    double dm2 = delta * deltaN * oldCount;
+
+    m1_ += deltaN;
+    m2_ += dm2;
+    m3_ += dm2 * deltaN * (count() - 2) - 3 * deltaN * oldM2;
+    m4_ += dm2 * deltaN2 * (1.0 * count() * count() - 3.0 * count() + 3) +
```

Comment:
@Yuhta I have made some changes here as well. Change `count() * (double)count()` to `1.0 * count() * count()` for the uniformity,  and change `3 * count()` to `3.0 * count()` to prevent overflow.